frontend/src/metabase/search/components/SearchFilterModal/filters/TypeFilter.tsx

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable react/prop-types */
+import { t } from "ttag";
+import { getTranslatedEntityName } from "metabase/nav/utils";
+import { Checkbox } from "metabase/ui";
+import { useSearchListQuery } from "metabase/common/hooks";
+import LoadingSpinner from "metabase/components/LoadingSpinner";
+import { SearchFilterView } from "metabase/search/components/SearchFilterModal/filters/SearchFilterView";
+
+import type { SearchFilterComponent } from "metabase/search/types";
+import { TypeCheckboxGroupWrapper } from "metabase/search/components/SearchFilterModal/filters/TypeFilter.styled";
+
+const EMPTY_SEARCH_QUERY = { models: "dataset", limit: 1 } as const;
+
+export const TypeFilter: SearchFilterComponent<"type"> = ({
+  value = [],
+  onChange,
+  "data-testid": dataTestId,
+}) => {
+  const { metadata, isLoading } = useSearchListQuery({
+    query: EMPTY_SEARCH_QUERY,
+  });
+
+  const availableModels = (metadata && metadata.available_models) ?? [];
```

Comment:
We'll want to handle a "No types" situation here - I'll consult with the PD/PM about this